persistence/sql/identity/persister_identity.go

```diff
@@ -680,3 +715,90 @@ func (p *Persister) injectTraitsSchemaURL(ctx context.Context, i *identity.Ident
 	i.SchemaURL = s.SchemaURL(p.r.Config().SelfPublicURL(ctx)).String()
 	return nil
 }
+
+type quotable interface {
+	Quote(key string) string
+}
+
+type node interface {
+	GetID() uuid.UUID
+	GetNID() uuid.UUID
+}
+
+func (p *IdentityPersister) update(ctx context.Context, v node, columnNames ...string) error {
+	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.update")
+	defer span.End()
+
+	c := p.GetConnection(ctx)
+	quoter, ok := c.Dialect.(quotable)
+	if !ok {
+		return errors.Errorf("store is not a quoter: %T", p.c.Store)
+	}
+
+	model := pop.NewModel(v, ctx)
+	tn := model.TableName()
+
+	cols := columns.Columns{}
+	if len(columnNames) > 0 && tn == model.TableName() {
+		cols = columns.NewColumnsWithAlias(tn, model.As, model.IDField())
+		cols.Add(columnNames...)
+	} else {
+		cols = columns.ForStructWithAlias(v, tn, model.As, model.IDField())
+	}
+
+	// #nosec
+	stmt := fmt.Sprintf("SELECT COUNT(id) FROM %s AS %s WHERE %s.id = ? AND %s.nid = ?",
+		quoter.Quote(model.TableName()),
+		model.Alias(),
+		model.Alias(),
+		model.Alias(),
+	)
```

Comment:
Is this safe and protected against SQL injection?

Reply:
Maybe provide an explanation for the `#nosec` statement (I'm assuming it suppresses a SQL injection warning)

Reply:
I've went over the whole codebase and cleaned up the `//#nosec` lint annotations everywhere.